components/org.wso2.carbon.consent.mgt.core/src/main/java/org/wso2/carbon/consent/mgt/core/ConsentManager.java

```diff
@@ -405,24 +405,105 @@ public Receipt getReceipt(String receiptId) throws ConsentManagementException {
         return receipt;
     }
 
+    /**
+     * This API is used to select the PIIController from List of registered PIIController. By default set the height
+     * priority one.
+     * @param piiControllers list of PIIControllers.
+     * @return selected PIIController.
+     */
+    protected PIIController getPIIController(List<PIIController> piiControllers) {
+
+        PIIController piiController = null;
+        if (piiControllers != null && piiControllers.size() > 0) {
+            piiController = piiControllers.get(piiControllers.size() - 1);
+        }
+        return piiController;
+    }
+
+    /**
+     * This API is used to select the ReceiptDAO from List of registered ReceiptDAO. By default set the height
+     * priority one.
+     * @param receiptDAOs list of ReceiptDAOs.
+     * @return selected ReceiptDAO.
+     */
+    protected ReceiptDAO getReceiptsDAO(List<ReceiptDAO> receiptDAOs) {
+
+        ReceiptDAO receiptDAO = null;
+        if (receiptDAOs != null && receiptDAOs.size() > 0) {
+            receiptDAO = receiptDAOs.get(receiptDAOs.size() - 1);
+        }
+        return receiptDAO;
+    }
+
+    /**
+     * This API is used to select the PIICategoryDAO from List of registered PIICategoryDAOs. By default set the height
+     * priority one.
+     * @param piiCategoryDAOs list of PIICategoryDAOs.
+     * @return selected PIICategoryDAO.
+     */
+    protected PIICategoryDAO getPiiCategoryDAO(List<PIICategoryDAO> piiCategoryDAOs) {
+
+        PIICategoryDAO piiCategoryDAO = null;
+        if (piiCategoryDAOs != null && piiCategoryDAOs.size() > 0) {
+            piiCategoryDAO = piiCategoryDAOs.get(piiCategoryDAOs.size() - 1);
+        }
+        return piiCategoryDAO;
+    }
+
+    /**
+     * This API is used to select the PurposeCategoryDAO from List of registered PurposeCategoryDAOs. By default set the
+     * height priority one.
+     * @param purposeCategoryDAOs list of PurposeCategoryDAOs.
+     * @return selected PurposeCategoryDAO.
+     */
+    protected PurposeCategoryDAO getPurposeCategoryDAO(List<PurposeCategoryDAO> purposeCategoryDAOs) {
+
+        PurposeCategoryDAO purposeCategoryDAO = null;
+        if (purposeCategoryDAOs != null && purposeCategoryDAOs.size() > 0) {
+            purposeCategoryDAO = purposeCategoryDAOs.get(purposeCategoryDAOs.size() - 1);
+        }
+        return purposeCategoryDAO;
+    }
+
+    /**
+     * This API is used to select the PurposeDAO from List of registered PurposeDAOs. By default set the
+     * height priority one.
+     * @param purposeDAOs list of PurposeDAOs.
+     * @return selected PurposeDAO.
+     */
+    protected PurposeDAO getPurposeDAO(List<PurposeDAO> purposeDAOs) {
+
+        PurposeDAO purposeDAO = null;
+        if (purposeDAOs != null && purposeDAOs.size() > 0) {
```

Comment:
Shall we use a util method to check this ?

Reply:
done